src/compile/problems.ts

```diff
@@ -216,11 +219,11 @@ export class KeyProblem extends Problem<KeyProblemKind> {
     mustBe = this.rule === "missing" ? "defined" : "extraneous"
 }
 
-export class RangeProblem extends Problem<Bound, SizedData> {
+export class RangeProblem extends Problem<Bound, NumericallyBoundableData> {
     readonly code = "range"
 
     get mustBe() {
-        return `${comparatorDescriptions[this.rule.comparator]} ${
+        return `${numericComparatorDescriptions[this.rule.comparator]} ${
```

Comment:
This would probably need to be adapted to include dates depending on the range, but lets not worry about that for now until I finish refactoring problems.